src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,67 +165,252 @@ private void InitializeDefaultAutoCompleteCandidates()
+ 
+                        var nodesFromAssembly = nodeModelSearchElements.Where(n => Path.GetFileNameWithoutExtension(n.Assembly).Equals(assemblyName));
+                        var matchingNode = nodesFromAssembly.FirstOrDefault(n => n.CreationName.Equals(fullName));
+                        if (matchingNode != null)
+                        {
+                            matchingNode.AutoCompletionNodeMachineLearningInfo.ConfidenceScore = item.Score;
+                            matchingNode.AutoCompletionNodeMachineLearningInfo.ViewConfidenceScoreRecentUse = true;
+                            matchingNode.AutoCompletionNodeMachineLearningInfo.IsByRecommendation = true;
+                            matchingResults.Add(matchingNode);
+                        }
+                    }
+                }
+
+                var results = GetViewModelForNodeSearchElements(matchingResults);
+
+                foreach (var result in results)
+                {
+                    if (result.Model.AutoCompletionNodeMachineLearningInfo.ConfidenceScore >= confidenceThreshold)
+                    {
+                        FilteredHighConfidenceResults = FilteredHighConfidenceResults.Append(result);
+                    }
+                    else {
+                        FilteredLowConfidenceResults = FilteredLowConfidenceResults.Append(result);
+                    }
+                }
+
+                // Show low confidence section if there are some results under threshold.
+                DisplayLowConfidence = FilteredLowConfidenceResults.Count() > 0;
+
+                // Show low confidence page when all results are under threshold. 
+                if (FilteredHighConfidenceResults.Count() <= 0)
+                {
+                    DisplayNoRecommendationsLowConfidence = true;
+                    NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteLowConfidenceTitle;
+                    NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteLowConfidenceMessage;
+                    LowConfidenceMessageAdditional = Resources.AutocompleteLowConfidenceMessageAditional;
+                    return;
+                }
+
+                // By default, show only the results which are above the threshold
+                FilteredResults = FilteredHighConfidenceResults;
+            }
+        }
+
+        private MLNodeAutoCompletionResponse GetMLNodeAutocompleteResults(string requestJSON)
+        {
+            MLNodeAutoCompletionResponse results = null;
+            var uri = DynamoUtilities.PathHelper.getServiceBackendAddress(this, "MLNodeAutocomplete");
+            var token = DynamoUtilities.PathHelper.getServiceConfigValues(this, "token");
+            var authProvider = dynamoViewModel.Model.AuthenticationManager.AuthProvider;
+
+            if (authProvider is IDSDKManager manager)
+            {
+                /*
+                var client = new RestClient(uri);
```

Comment:
Getting 403 forbidden error when using the RestSharp API. Will undo this code, once that is unblocked.

Reply:
Changed this to use the RestSharp API and setting the IDSDK token in header if the user is logged in.